File: CourseWebAPI/Infrastuctures/Models/InstructorModel.cs
using CourseWebAPI.Infrastuctures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace CourseWebAPI.Models
{
    public class InstructorModel 
    {
        [Required]
        [MaxLength(50)]
        public virtual string LastName { get; set; }

        [Required]
        [MaxLength(500)]
        public virtual string FirstMidName { get; set; }
        public virtual DateTime HireDate { get; set; }
        public virtual string OfficeLocation {get; set;}
    }
}

File: course-management-client/src/app/feature-components/instructors/instructor.model.ts
export interface IInstructorParams {}

export interface IInstructorList {
  id: number;
  lastName: string;
  firstMidName: string;
  hireDate: string;
  departments: string[];
  officeLocation: string;
  courses: string[];
}

export interface IInstructor {
  id: number;
  lastName: string;
  firstMidName: string;
  hireDate: string;
  officeLocation: string;
}
